domain/src/main/java/life/qbic/domain/user/User.java

```diff
@@ -0,0 +1,156 @@
+   * created.
+   *
+   * <p>The object instance won't hold a reference to the original password char array, after it
+   * has been encrypted.
+   *
+   * @param fullName          the full name of the user
+   * @param emailAddress             the emailAddress address of the user
+   * @param encryptedPassword the encrypted password of the new user
+   * @return the new user
+   * @since 1.0.0
+   */
+  public static User create(FullName fullName, EmailAddress emailAddress, EncryptedPassword encryptedPassword) {
+    String uuid = String.valueOf(UUID.randomUUID());
+    var user = new User(fullName);
+    user.setEmail(emailAddress);
+    user.setId(uuid);
+    user.setEncryptedPassword(encryptedPassword);
+    user.active = false;
+
+    return user;
+  }
+
+  /**
+   * Recreates an instance of a user object, for example when loading user data from the persistence
+   * layer.
+   *
+   * @param encryptedPassword the encrypted password
+   * @param fullName          the full name
+   * @param emailAddress             the emailAddress
+   * @return an object instance of the user
+   * @since 1.0.0
+   */
+  protected static User of(EncryptedPassword encryptedPassword, FullName fullName, EmailAddress emailAddress) {
+    var user = new User(fullName);
+    user.setEmail(emailAddress);
+    user.setEncryptedPassword(encryptedPassword);
+    return user;
+  }
+
+  private User(FullName fullName) {
+    this.fullName = fullName;
+  }
+
+  @Override
+  public String toString() {
+    return "User{" +
+        "id='" + id + '\'' +
+        ", fullName=" + fullName +
+        ", emailAddress=" + emailAddress +
+        ", encryptedPassword=" + encryptedPassword +
+        ", active=" + active +
+        '}';
+  }
+
+  private void setEncryptedPassword(EncryptedPassword encryptedPassword) {
+    this.encryptedPassword = encryptedPassword;
+  }
+
+  /**
+   * Get access to the encrypted password
+   *
+   * @return the password
+   * @since 1.0.0
+   */
+  public EncryptedPassword getEncryptedPassword() {
+    return this.encryptedPassword;
+  }
+
+  /**
+   * Sets the emailAddress address for the current user.
+   *
+   * @param emailAddress the emailAddress address of the user
+   * @since 1.0.0
+   */
+  private void setEmail(EmailAddress emailAddress) {
+    this.emailAddress = emailAddress;
+  }
+
+  private void setId(String id) {
+    this.id = id;
+  }
+
+  public String getId() {
+    return this.id;
+  }
+
+  public EmailAddress getEmail() {
+    return this.emailAddress;
+  }
+
+  public FullName getFullName() {
+    return this.fullName;
+  }
+
+  /**
+   * Checks if a given password is correct for a user
+   *
+   * @param rawPassword EncryptedPassword that is being validated
+   * @return true, if the given password is correct for the user
+   */
+  public Boolean checkPassword(char[] rawPassword) {
+    return Objects.equals(
+        PasswordEncryptionPolicy.create().encrypt(rawPassword), encryptedPassword.hash());
+  }
```

Comment:
I do not think we need this method as this is handled by Spring Security (using the password policy).

Reply:
On the one hand: YAGNI
On the other hand checking the provided password might be useful if a user wants to make changes within his profile.

Reply:
I see your point though and will remove it for now 👍